crates/fuel-core/src/schema/coins.rs

```diff
@@ -143,6 +157,49 @@ pub struct ExcludeInput {
     messages: Vec<Nonce>,
 }
 
+pub struct ExcludedKeysAsBytes {
+    coins: Vec<CoinOrMessageIdBytes>,
+    messages: Vec<CoinOrMessageIdBytes>,
+}
+
+// The part of the `CoinsToSpendIndexKey` which is used to identify the coin or message in the
+// OnChain database. We could consider using `CoinId`, but we do not need to re-create
+// neither the `UtxoId` nor `Nonce` from the raw bytes.
+#[derive(PartialEq)]
+pub(crate) enum CoinOrMessageIdBytes {
+    Coin([u8; COIN_FOREIGN_KEY_LEN]),
+    Message([u8; MESSAGE_FOREIGN_KEY_LEN]),
+}
+
+impl CoinOrMessageIdBytes {
+    pub(crate) fn from_utxo_id(utxo_id: &fuel_tx::UtxoId) -> Self {
+        Self::Coin(utxo_id_to_bytes(utxo_id))
+    }
+
+    pub(crate) fn from_nonce(nonce: &fuel_types::Nonce) -> Self {
+        let mut arr = [0; MESSAGE_FOREIGN_KEY_LEN];
+        arr[0..32].copy_from_slice(nonce.as_ref());
```

Comment:
```suggestion
        arr.copy_from_slice(nonce.as_ref());
```

Reply:
Updated in 3ee2a1a9e1ab78c3551b3ab9972b90c42b7e6105